doc/development/deprecations.rst

```diff
@@ -132,6 +132,27 @@ Pending deprecations
   - Deprecated in v0.32
   - Will be removed in v0.33
 
+* The following decorator syntax for batch transforms has been deprecated:
+
+  .. code-block:: python
+
+      @transform_fn(*transform_args)
+      @qml.qnode(dev)
+      def circuit():
+          ...
+
+  Please call the batch transform directly using ``circuit = transform_fn(circuit, *transform_args)``,
+  or use ``functools.partial``:
+
+  .. code-block:: python
+
+      @functools.partial(transform_fn, *transform_args)
+      @qml.qnode(dev)
+      def circuit():
+          ...
+
+  - Deprecated in v0.32
```

Comment:
```suggestion
  - Deprecated in v0.33
```
I don't think all transforms will be in by the release

Reply:
Sorry we can deprecate it without having the transform rework in.

Reply:
Though doing this deprecation in 0.33 might be a nice idea, given that we'll have to fix lots of demos and potentially some plugins.

Reply:
I'm fine with either. Although if we deprecate in 0.33, then I need to separate out the changes to support the old decorator syntax for the new transform API.

Reply:
Deprecate it just on the transforms that have been ported?

Reply:
Oh I think that works. Then I can remove the warning for the old batch transforms class.

Reply:
Note that we aren't mixing old and new transform APIs in 0.32, we'll go for a full switchover in 0.33.

Reply:
Ok, 0.33 it is 👍